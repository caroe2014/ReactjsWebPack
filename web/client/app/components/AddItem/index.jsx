// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import React, { Component } from 'react';
import { Flex } from 'rebass';
import styled from 'styled-components';
import IconButton from 'web/client/app/components/IconButton';

const Input = styled(Flex)`
  padding: 0 5px;
  color: ${props => props.theme.colors.textGrey};
`;

const TileContainer = styled(Flex)`
  border: 0.5px solid ${props => props.theme.colors.textGrey};
  border-radius: 15px;
  padding: 0.2em 0.75em;
  height: 32px;
`;

const Button = styled(IconButton)`
  font-size: ${props => props.theme.fontSize.sm};
  color: ${props => props.theme.colors.textGrey};
  &:focus{
   box-shadow: none;
   outline: none;
  }

`;

class AddItem extends Component {
  constructor (props) {
    super(props);
    this.state = {
      value: this.props.value || 1
    };

    this.handleChange = this.handleChange.bind(this);
    this.handleInputChange = this.handleInputChange.bind(this);
  }

  doDecrement = () => {
    const { min } = this.props;
    if (!min || min < this.state.value) {
      let value = this.state.value - 1;
      this.setState({ value });
      this.handleChange(value);
    }
  }

  doIncrement = () => {
    const { max } = this.props;
    if (!max || max > this.state.value) {
      let value = this.state.value + 1;
      this.setState({ value });
      this.handleChange(value);
    }
  }

  handleChange (value) {
    const { onInputChange } = this.props;
    if (onInputChange) {
      onInputChange(value);
    }
  }

  handleInputChange (e) {
    const target = e.target;
    const value = parseInt(target.value);
    this.setState({ value });
    this.handleChange(value);
  }

  render () {
    const { width } = this.props;
    return (
      <TileContainer width={width ? width + 'px' : 'auto'} >
        <Flex justifyContent='space-around' alignItems='center' width='100%'>
          <Button
            className='decrement-button'
            aria-label='decrement'
            onClick={this.doDecrement}
            iconClassName='fa fa-minus'
            role='button'
          />
          <Input>
            {this.state.value}
          </Input>
          <Button
            className='increment-button'
            aria-label='increment'
            onClick={this.doIncrement}
            iconClassName='fa fa-plus'
            role='button'
          />
        </Flex>
      </TileContainer>
    );
  }
}

export default AddItem;
