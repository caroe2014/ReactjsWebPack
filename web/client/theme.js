// Copyright Â© 2018-2019 Agilysys NV, LLC.  All Rights Reserved.  Confidential Information of Agilysys NV, LLC.

import { css, injectGlobal } from 'styled-components';

const theme = {
  colors: {
    black: '#000000',
    navigationMenu: '#48a942',
    navigationMenuHover: 'cornflowerblue',
    primary: '#347CFF',
    primaryButton: '#347CFF',
    loginField: '#AAA7AC',
    placeHolder: '#909090',
    loginBoxColor: '#434149',
    open: '#48a942',
    close: 'red',
    footer: '#48a942',
    textGrey: '#505050',
    paymentSuccessTitle: '#505050',
    inputBottomBorder: 'lightgray',
    backgroundGrey: '#E6E6E6',
    text: 'black',
    primaryText: '#383838',
    secondaryText: '#a6a6a6',
    hint: '#a6a6a6',
    light: 'white',
    error: '#cc0000',
    tile: '#505050',
    addToCart: '#347CFF',
    itemTile: '#3C474C0',
    darkGrey: '#464646',
    itemBorder: 'lightgrey',
    addOnTab: '#347CFF',
    validationError: '#cc0000',
    loginBackground: '#303030',
    loginFormBackground: '#484848',
    disableBackground: '#E4E4E4',
    addonBg: '#EEEEEE',
    glutenFree: '#8d7356',
    spicy: '#ed1c24',
    vegan: '#0d4a22',
    vegetarian: '#0d4a22',
    new: '#ff6000',
    healthy: '#359919',
    contextBackground: '#f0f0f3',
    loader: '#ababab',
    disableButton: '#D1D1D1'
  },
  fontFamily: 'Roboto',
  receiptFontFamily: 'Roboto',
  fontSize: {
    xl: '2.4rem',
    lg: '1.8rem',
    md: '1.3rem',
    nm: '1rem',
    sm: '0.75rem'
  },
  fontSizeFromProps: (props) => {
    if (props.xl) return props.theme.fontSize.xl;
    if (props.lg) return props.theme.fontSize.lg;
    if (props.md) return props.theme.fontSize.md;
    if (props.nm) return props.theme.fontSize.nm;
    if (props.sm) return props.theme.fontSize.sm;
    return 'inherit';
  },
  spacing: {
    xl: '8em',
    lg: '4em',
    md: '2em',
    nm: '1em',
    sm: '0.5em',
    none: '0em'
  },
  mediaBreakpoints: {
    mobile: (...args) => css`@media (max-width: 32em) {${css(...args)}}`,
    tablet: (...args) => css`@media (max-width: 72em) {${css(...args)}}`,
    desktop: (...args) => css`@media (min-width: 72em) {${css(...args)}}`
  },
  shadow: '#cfcfcf 0.1px 0.1px 3px 0px',
  nav: {
    height: '70px'
  }
};

injectGlobal`
  .agilysys-icon-new {
    color: ${theme.colors.new}
  }
  .agilysys-icon-healthy {
    color: ${theme.colors.healthy}
  }
  .agilysys-icon-vegetarian {
    color: ${theme.colors.vegetarian}
  }
  .agilysys-icon-vegan {
    color: ${theme.colors.vegan}
  }
  .agilysys-icon-spicy {
    color: ${theme.colors.spicy}
  }
  .agilysys-icon-gluten-free {
    color: ${theme.colors.glutenFree}
  }
  @media print {
    @-moz-document url-prefix() {
      .appContainer, .appContainer>div, .appContainer .parent, .appContainer .BottomContainer {
        display: block !important;
      }
    }
  }
`;

export default theme;
