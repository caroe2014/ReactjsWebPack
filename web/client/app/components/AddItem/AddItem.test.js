import React from 'react';
import {
  mount,
  shallow,
  configure
} from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import 'jest-styled-components';
import theme from 'web/client/theme';
import { ThemeProvider } from 'styled-components';

import AddItem from '.';
configure({
  adapter: new Adapter()
});

describe('IconButton', () => {

  it('does a snapshot test', () => {
    const wrapper = mount(<ThemeProvider theme={theme}><AddItem theme={theme} min={1}/></ThemeProvider>);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should render without throwing an error', () => {
    expect(mount(<ThemeProvider theme={theme}><AddItem theme={theme} min={1}/></ThemeProvider>).exists()).toBe(true);
  });

  it('should call doDecrement with spy', () => {
    let wrapper = shallow(<AddItem theme={theme} min={1}/>);
    const spy = jest.spyOn(wrapper.instance(), 'doDecrement');
    wrapper.instance().forceUpdate();
    wrapper.find('.decrement-button').simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });

  it('should call doDecrement with spy with value set', () => {
    let wrapper = shallow(<AddItem theme={theme} min={1}/>);
    wrapper.setState({ value: 3 });
    const spy = jest.spyOn(wrapper.instance(), 'doDecrement');
    wrapper.instance().forceUpdate();
    wrapper.find('.decrement-button').simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });

  it('should call doIncrement with spy', () => {
    let wrapper = shallow(<AddItem theme={theme} max={10}/>);
    const spy = jest.spyOn(wrapper.instance(), 'doIncrement');
    wrapper.instance().forceUpdate();
    wrapper.find('.increment-button').simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });

  it('should call doIncrement with spy with value set', () => {
    let wrapper = shallow(<AddItem theme={theme} max={5}/>);
    wrapper.setState({ value: 10 });
    const spy = jest.spyOn(wrapper.instance(), 'doIncrement');
    wrapper.instance().forceUpdate();
    wrapper.find('.increment-button').simulate('click');
    expect(spy).toHaveBeenCalled();
    spy.mockClear();
  });
});
