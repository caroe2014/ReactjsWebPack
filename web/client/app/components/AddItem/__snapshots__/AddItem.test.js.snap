// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton does a snapshot test 1`] = `
.c2 {
  box-sizing: border-box;
  width: auto;
}

.c4 {
  box-sizing: border-box;
  width: 100%;
}

.c9 {
  box-sizing: border-box;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: space-around;
  -webkit-justify-content: space-around;
  -ms-flex-pack: space-around;
  justify-content: space-around;
}

.c7 {
  font-family: inherit;
  -webkit-font-smoothing: antialiased;
  display: inline-block;
  vertical-align: middle;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  font-size: 14px;
  font-weight: bold;
  line-height: 1.1428571428571428;
  margin: 0px;
  padding-left: 16px;
  padding-right: 16px;
  padding-top: 8px;
  padding-bottom: 8px;
  color: white;
  background-color: blue;
  border-radius: 9999px;
  border: 0;
}

.c7:disabled {
  opacity: 0.25;
}

.c7:focus {
  outline: none;
  box-shadow: 0 0 0 2px blue;
}

.c6 {
  background-color: transparent;
  color: #a6a6a6;
  cursor: pointer;
  font-size: 1.3rem;
  padding: 0;
}

.c6:focus {
  outline: none;
  box-shadow: none;
}

.c8 {
  padding: 0 5px;
  color: #505050;
}

.c0 {
  border: 0.5px solid #505050;
  border-radius: 15px;
  padding: 0.2em 0.75em;
  height: 32px;
}

.c5 {
  font-size: 0.75rem;
  color: #505050;
}

.c5:focus {
  box-shadow: none;
  outline: none;
}

<ThemeProvider
  theme={
    Object {
      "colors": Object {
        "addOnTab": "#347CFF",
        "addToCart": "#347CFF",
        "addonBg": "#EEEEEE",
        "backgroundGrey": "#E6E6E6",
        "black": "#000000",
        "close": "red",
        "contextBackground": "#f0f0f3",
        "darkGrey": "#464646",
        "disableBackground": "#E4E4E4",
        "disableButton": "#D1D1D1",
        "error": "#cc0000",
        "footer": "#48a942",
        "glutenFree": "#8d7356",
        "healthy": "#359919",
        "hint": "#a6a6a6",
        "inputBottomBorder": "lightgray",
        "itemBorder": "lightgrey",
        "itemTile": "#3C474C0",
        "light": "white",
        "loader": "#ababab",
        "loginBackground": "#303030",
        "loginBoxColor": "#434149",
        "loginField": "#AAA7AC",
        "loginFormBackground": "#484848",
        "navigationMenu": "#48a942",
        "navigationMenuHover": "cornflowerblue",
        "new": "#ff6000",
        "open": "#48a942",
        "paymentSuccessTitle": "#505050",
        "placeHolder": "#909090",
        "primary": "#347CFF",
        "primaryButton": "#347CFF",
        "primaryText": "#383838",
        "secondaryText": "#a6a6a6",
        "spicy": "#ed1c24",
        "text": "black",
        "textGrey": "#505050",
        "tile": "#505050",
        "validationError": "#cc0000",
        "vegan": "#0d4a22",
        "vegetarian": "#0d4a22",
      },
      "fontFamily": "Roboto",
      "fontSize": Object {
        "lg": "1.8rem",
        "md": "1.3rem",
        "nm": "1rem",
        "sm": "0.75rem",
        "xl": "2.4rem",
      },
      "fontSizeFromProps": [Function],
      "mediaBreakpoints": Object {
        "desktop": [Function],
        "mobile": [Function],
        "tablet": [Function],
      },
      "nav": Object {
        "height": "70px",
      },
      "receiptFontFamily": "Roboto",
      "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
      "spacing": Object {
        "lg": "4em",
        "md": "2em",
        "nm": "1em",
        "none": "0em",
        "sm": "0.5em",
        "xl": "8em",
      },
    }
  }
>
  <AddItem
    min={1}
    theme={
      Object {
        "colors": Object {
          "addOnTab": "#347CFF",
          "addToCart": "#347CFF",
          "addonBg": "#EEEEEE",
          "backgroundGrey": "#E6E6E6",
          "black": "#000000",
          "close": "red",
          "contextBackground": "#f0f0f3",
          "darkGrey": "#464646",
          "disableBackground": "#E4E4E4",
          "disableButton": "#D1D1D1",
          "error": "#cc0000",
          "footer": "#48a942",
          "glutenFree": "#8d7356",
          "healthy": "#359919",
          "hint": "#a6a6a6",
          "inputBottomBorder": "lightgray",
          "itemBorder": "lightgrey",
          "itemTile": "#3C474C0",
          "light": "white",
          "loader": "#ababab",
          "loginBackground": "#303030",
          "loginBoxColor": "#434149",
          "loginField": "#AAA7AC",
          "loginFormBackground": "#484848",
          "navigationMenu": "#48a942",
          "navigationMenuHover": "cornflowerblue",
          "new": "#ff6000",
          "open": "#48a942",
          "paymentSuccessTitle": "#505050",
          "placeHolder": "#909090",
          "primary": "#347CFF",
          "primaryButton": "#347CFF",
          "primaryText": "#383838",
          "secondaryText": "#a6a6a6",
          "spicy": "#ed1c24",
          "text": "black",
          "textGrey": "#505050",
          "tile": "#505050",
          "validationError": "#cc0000",
          "vegan": "#0d4a22",
          "vegetarian": "#0d4a22",
        },
        "fontFamily": "Roboto",
        "fontSize": Object {
          "lg": "1.8rem",
          "md": "1.3rem",
          "nm": "1rem",
          "sm": "0.75rem",
          "xl": "2.4rem",
        },
        "fontSizeFromProps": [Function],
        "mediaBreakpoints": Object {
          "desktop": [Function],
          "mobile": [Function],
          "tablet": [Function],
        },
        "nav": Object {
          "height": "70px",
        },
        "receiptFontFamily": "Roboto",
        "shadow": "#cfcfcf 0.1px 0.1px 3px 0px",
        "spacing": Object {
          "lg": "4em",
          "md": "2em",
          "nm": "1em",
          "none": "0em",
          "sm": "0.5em",
          "xl": "8em",
        },
      }
    }
  >
    <Styled(Flex)
      width="auto"
    >
      <Flex
        className="c0"
        width="auto"
      >
        <Box
          className="c0 c1"
          width="auto"
        >
          <div
            className="c0 c1 c2"
            width="auto"
          >
            <Flex
              alignItems="center"
              justifyContent="space-around"
              width="100%"
            >
              <Box
                alignItems="center"
                className="c3"
                justifyContent="space-around"
                width="100%"
              >
                <div
                  className="c3 c4"
                  width="100%"
                >
                  <Styled(IconButton)
                    aria-label="decrement"
                    className="decrement-button"
                    iconClassName="fa fa-minus"
                    onClick={[Function]}
                    role="button"
                  >
                    <IconButton
                      aria-label="decrement"
                      className="decrement-button c5"
                      iconClassName="fa fa-minus"
                      onClick={[Function]}
                      role="button"
                    >
                      <Styled(ButtonCircle)
                        aria-label="decrement"
                        className="decrement-button c5"
                        onClick={[Function]}
                        role="button"
                      >
                        <ButtonCircle
                          aria-label="decrement"
                          bg="blue"
                          border={0}
                          borderRadius={9999}
                          className="decrement-button c5 c6"
                          color="white"
                          extend={[Function]}
                          fontSize={1}
                          fontWeight="bold"
                          is="button"
                          lineHeight={1.1428571428571428}
                          m={0}
                          onClick={[Function]}
                          px={3}
                          py={2}
                          role="button"
                        >
                          <ForwardRef
                            aria-label="decrement"
                            bg="blue"
                            border={0}
                            borderRadius={9999}
                            className="decrement-button c5 c6 "
                            color="white"
                            extend={[Function]}
                            fontSize={1}
                            fontWeight="bold"
                            is="button"
                            lineHeight={1.1428571428571428}
                            m={0}
                            onClick={[Function]}
                            px={3}
                            py={2}
                            role="button"
                          >
                            <Button
                              aria-label="decrement"
                              bg="blue"
                              border={0}
                              borderRadius={9999}
                              className="decrement-button c5 c6 "
                              color="white"
                              fontSize={1}
                              fontWeight="bold"
                              is="button"
                              lineHeight={1.1428571428571428}
                              m={0}
                              onClick={[Function]}
                              px={3}
                              py={2}
                              role="button"
                            >
                              <ForwardRef
                                aria-label="decrement"
                                bg="blue"
                                border={0}
                                borderRadius={9999}
                                className="decrement-button c5 c6 c7"
                                color="white"
                                fontSize={1}
                                fontWeight="bold"
                                is="button"
                                lineHeight={1.1428571428571428}
                                m={0}
                                onClick={[Function]}
                                px={3}
                                py={2}
                                role="button"
                              >
                                <button
                                  aria-label="decrement"
                                  className="decrement-button c5 c6 c7"
                                  onClick={[Function]}
                                  role="button"
                                >
                                  <i
                                    className="fa fa-minus"
                                  />
                                </button>
                              </ForwardRef>
                            </Button>
                          </ForwardRef>
                        </ButtonCircle>
                      </Styled(ButtonCircle)>
                    </IconButton>
                  </Styled(IconButton)>
                  <Styled(Flex)>
                    <Flex
                      className="c8"
                    >
                      <Box
                        className="c8 c1"
                      >
                        <div
                          className="c8 c1 c9"
                        >
                          1
                        </div>
                      </Box>
                    </Flex>
                  </Styled(Flex)>
                  <Styled(IconButton)
                    aria-label="increment"
                    className="increment-button"
                    iconClassName="fa fa-plus"
                    onClick={[Function]}
                    role="button"
                  >
                    <IconButton
                      aria-label="increment"
                      className="increment-button c5"
                      iconClassName="fa fa-plus"
                      onClick={[Function]}
                      role="button"
                    >
                      <Styled(ButtonCircle)
                        aria-label="increment"
                        className="increment-button c5"
                        onClick={[Function]}
                        role="button"
                      >
                        <ButtonCircle
                          aria-label="increment"
                          bg="blue"
                          border={0}
                          borderRadius={9999}
                          className="increment-button c5 c6"
                          color="white"
                          extend={[Function]}
                          fontSize={1}
                          fontWeight="bold"
                          is="button"
                          lineHeight={1.1428571428571428}
                          m={0}
                          onClick={[Function]}
                          px={3}
                          py={2}
                          role="button"
                        >
                          <ForwardRef
                            aria-label="increment"
                            bg="blue"
                            border={0}
                            borderRadius={9999}
                            className="increment-button c5 c6 "
                            color="white"
                            extend={[Function]}
                            fontSize={1}
                            fontWeight="bold"
                            is="button"
                            lineHeight={1.1428571428571428}
                            m={0}
                            onClick={[Function]}
                            px={3}
                            py={2}
                            role="button"
                          >
                            <Button
                              aria-label="increment"
                              bg="blue"
                              border={0}
                              borderRadius={9999}
                              className="increment-button c5 c6 "
                              color="white"
                              fontSize={1}
                              fontWeight="bold"
                              is="button"
                              lineHeight={1.1428571428571428}
                              m={0}
                              onClick={[Function]}
                              px={3}
                              py={2}
                              role="button"
                            >
                              <ForwardRef
                                aria-label="increment"
                                bg="blue"
                                border={0}
                                borderRadius={9999}
                                className="increment-button c5 c6 c7"
                                color="white"
                                fontSize={1}
                                fontWeight="bold"
                                is="button"
                                lineHeight={1.1428571428571428}
                                m={0}
                                onClick={[Function]}
                                px={3}
                                py={2}
                                role="button"
                              >
                                <button
                                  aria-label="increment"
                                  className="increment-button c5 c6 c7"
                                  onClick={[Function]}
                                  role="button"
                                >
                                  <i
                                    className="fa fa-plus"
                                  />
                                </button>
                              </ForwardRef>
                            </Button>
                          </ForwardRef>
                        </ButtonCircle>
                      </Styled(ButtonCircle)>
                    </IconButton>
                  </Styled(IconButton)>
                </div>
              </Box>
            </Flex>
          </div>
        </Box>
      </Flex>
    </Styled(Flex)>
  </AddItem>
</ThemeProvider>
`;
